playwright/page-objects/ocm-roles-access-page.ts

```diff
@@ -0,0 +1,69 @@
+import { Page, Locator, expect } from '@playwright/test';
+import { BasePage } from './base-page';
+
+/**
+ * OCM Roles and Access page object for Playwright tests
+ */
+export class OCMRolesAndAccessPage extends BasePage {
+  constructor(page: Page) {
+    super(page);
+  }
+
+  accessControlTabButton(): Locator {
+    return this.page.getByRole('tab', { name: 'Access control' });
+  }
+
+  async assertUrlIncludes(fragment: string): Promise<void> {
+    await expect(this.page).toHaveURL(new RegExp(fragment));
+  }
+
+  grantRoleButton(): Locator {
+    return this.page.getByRole('button', { name: 'Grant role' });
+  }
+
+  OCMRolesAndAccessTable(): Locator {
+    return this.page.locator('table[aria-label="OCM Roles and Access"]');
+  }
+
+  ocmRoleAndAccessDialog(): Locator {
+    return this.page.locator('#ocm-roles-access-dialog');
+  }
+
+  grantRoleUserInput(): Locator {
+    return this.page.locator('input[type="text"]:not([aria-label="Date picker"])').last();
+  }
+
+  userInputError(): Locator {
+    return this.page.locator(
+      '[class*="helper-text"], [class*="form__helper-text"], [class*="error"]',
```

Comment:
This one seems to be overly broad and might match unrelated elements. Kindly check